### PR TITLE
feat(vault-monitoring): make bearerTokenFile optional

### PR DESCRIPTION
**Pull request by pree**
# Description

When using an unauthenticated access service for monitoring we don't need the bearerTokenFile.

# Issues

# Checklist

* [ ] This PR contains a description of the changes I'm making
* [ ] I updated the version in Chart.yaml
* [ ] I updated the changelog with an `artifacthub.io/changes` annotation in `Chart.yaml`, check the [example](docs/development.md#Changelog) in the documentation.
* [ ] I updated applicable README.md files using  `pre-commit run`
* [ ] I documented any high-level concepts I'm introducing in `docs/`
* [ ] CI is currently green and this is ready for review
* [ ] I am ready to test changes after they are applied and released

### CODE REVIEW

File: charts/vault-monitoring/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -24,7 +24,9 @@ spec:
       port: {{ .Values.vault.portName }}
       scheme: {{ .Values.vault.scheme }}
       scrapeTimeout: {{ .Values.vault.serviceMonitor.scrapeTimeout }}
-      bearerTokenFile: '/etc/prometheus/config_out/.vault-token'
+{{- if .Values.vault.serviceMonitor.bearerToken.enabled }}
```
Comment (tongpu) [inline charts/vault-monitoring/templates/servicemonitor.yaml:27]:
```suggestion
      {{- if .Values.vault.serviceMonitor.bearerToken.enabled }}
```

File: charts/vault-monitoring/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -24,7 +24,9 @@ spec:
...
+      bearerTokenFile: {{ .Values.vault.serviceMonitor.bearerToken.path }}
+{{- end }}
```
Comment (tongpu) [inline charts/vault-monitoring/templates/servicemonitor.yaml:29]:
```suggestion
      {{- end }}
```

**Comment by eyenx** [comment]
Can you annotate thechanges to the Chart.yaml

**Comment by pree** [comment]
I added the changes, but unsure if this is needed as we can use the `serviceMonitor` from `vault-helm`. Externally hosted Vaults should never be contacted unauthenticated, so we might not want to merge this in.

File: charts/vault-monitoring/values.yaml [hunk 3]
```diff
@@ -39,11 +39,7 @@ vault:
     labels: {}
     # -- scrapeTimeout of the serviceMonitor
     scrapeTimeout: 30s
-    bearerToken:
-      # -- enable bearerToken authentication
-      enabled: true
-      # -- path to the bearerToken
-      path: '/etc/prometheus/config_out/.vault-token'
+    # -- Specify a bearerTokenFile for authentication. Keep empty for Unauthenticated access
```
Comment (hairmare) [inline charts/vault-monitoring/values.yaml:42]:
```suggestion
    # -- Specify a bearerTokenFile for authentication. Keep empty for Unauthenticated access
    bearerToken: ""
```

File: charts/vault-monitoring/values.yaml [hunk 4]
```diff
@@ -40,6 +40,7 @@ vault:
     # -- scrapeTimeout of the serviceMonitor
     scrapeTimeout: 30s
     # -- Specify a bearerTokenFile for authentication. Keep empty for Unauthenticated access
+    bearerTokenFile: '/etc/prometheus/config_out/.vault-token'
```
Comment (hairmare) [inline charts/vault-monitoring/values.yaml:43]: do you want the default to be "use a bearer token"? if so, then this is fine, if not, then an `""` might be preferential

**Review by hairmare — APPROVED** [review]

File: charts/vault-monitoring/values.yaml (hunk 4 above)
Comment (pree) [reply charts/vault-monitoring/values.yaml:43]: Yes, as it's highly recommended to secure the connection either by using the bearer token or securing the endpoint with other technologies (like network policies). Also, this way it's not a breaking change.

**Review by eyenx — APPROVED** [review]

**Pull request merged**